server/src/com/thoughtworks/go/config/update/DeletePipelineConfigCommand.java

```diff
@@ -50,6 +50,22 @@ public void update(CruiseConfig cruiseConfig) throws Exception {
 
     @Override
     public boolean isValid(CruiseConfig preprocessedConfig) {
+        for (PipelineConfig pipeline : preprocessedConfig.getAllPipelineConfigs()) {
+            if(pipeline.materialConfigs().hasDependencyMaterial(pipelineConfig)){
+                Localizable.CurryableLocalizable message = LocalizedMessage.string("CANNOT_DELETE_PIPELINE_USED_AS_MATERIALS", pipelineConfig.name(), pipeline.name());
+                this.result.unprocessableEntity(message);
+                return false;
+            }
+        }
+
+        for (EnvironmentConfig environment : goConfigService.getEnvironments()) {
```

Comment:
Please change this to `preprocessedConfig.getEnvironments()` here instead of asking the same from service